src/components/playback/DrawManager.js

```diff
@@ -0,0 +1,211 @@
+import EventEmitter from 'event-emitter'
+import allOff from 'event-emitter/all-off'
+import clamp from 'lodash/clamp'
+import last from 'lodash/last'
+
+const WIDTH = 1920
+const HEIGHT = 1080
+
+export default class DrawManager {
+  constructor (svg) {
+    this.svg = d3.select(svg)
+    this.lineFunction = d3.svg.line().x(d => d.x * WIDTH).y(d => d.y * HEIGHT).interpolate('cardinal')
+
+    this.lineCursor = 0
+    this.needsRedraw = false
+    this.parsedLines = []
+    this.pointCursor = 0
+    this.position = 0
+  }
+
+  prepare (stream, streamData, colors) {
+    this.duration = stream.duration * 1000
+    this.lines = streamData.split('\n').filter(l => l)
+    this.colors = colors
+
+    this._setUpLine()
+    this.emit('READY')
+  }
+
+  play () {
+    this.playing = true
+    this._playFrom(this.position)
+    this.emit('POSITION_CHANGE', this.position, true)
+    this.emit('PLAY')
+  }
+
+  pause () {
+    this.playing = false
+    this._cancelNextFrame()
+    this.emit('PAUSE')
+  }
+
+  toggle () {
+    this.playing ? this.pause() : this.play()
+  }
+
+  stop () {
+    allOff(this)
+  }
+
+  setPosition (position) {
+    const wasPlaying = this.playing
+    if (this.playing) this.pause()
+
+    if (position <= this.position) {
+      this.needsRedraw = true
+    }
+
+    this.position = position
+    this.emit('POSITION_CHANGE', this.position)
+
+    this._draw()
+
+    if (wasPlaying) this.play()
+  }
+
+  _playFrom (offset) {
+    let start
+
+    const initialPosition = this.position
+
+    const frame = (now) => {
+      if (!start) start = now
+
+      this.position = clamp(initialPosition + now - start, 0, this.duration)
+      this.emit('POSITION_CHANGE', this.position)
+
+      if (this._draw()) {
```

Comment:
`_draw()` returns false when there's nothing left to draw (i.e., once the stream is done playing).